lldb/tools/lldb-dap/src-ts/debug-adapter-factory.ts

```diff
@@ -28,6 +23,70 @@ export class LLDBDapDescriptorFactory
     return true;
   }
 
+  static async findDAPExecutable(): Promise<string | undefined> {
```

Comment:
This function is really not complex but a bit long. Can you split it into smaller chunks?